src/Microsoft.PowerApps.TestEngine/TestEngineEvents.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerApps.TestEngine.Config;
+
+namespace Microsoft.PowerApps.TestEngine
+{
+    /// <summary>
+    /// Handles all major Test Engine events using console output
+    /// </summary>
+    public class TestEngineEvents : ITestEngineEvents
+    {
+        public void SetNumberOfTotalCases(int numCases)
+        {
```

Comment:
Previously you were doing `Console.WriteLine` calls for these so the console app would have meaningful output. Do you want to do the same here as the default behavior?

Reply:
I'm thinking that we don't make the service optional, so that a consumer knows they have to provide an event handler in order to make the program work. Handling the events is a core part of Test Engine. I think that leaving it blank allows us to make it required but also not have anything logged by default.

Reply:
Sounds good, I defer to y'all on the expectations around behavior.